src/engine.rs

```diff
@@ -350,13 +351,19 @@ where
     }
 
     /// Dumps all operations.
-    #[allow(unused_variables)]
-    pub fn dump_with_file_builder(
-        path: &Path,
-        raft_groups: &[u64],
-        file_builder: Arc<B>,
-    ) -> Result<Vec<LogItem>> {
-        todo!()
+    pub fn dump_with_file_builder(path: &Path, file_builder: Arc<B>) -> Result<LogItemReader<B>> {
```

Comment:
Better implement https://doc.rust-lang.org/std/iter/trait.Iterator.html for `LogItemReader`

Reply:
And add an unit test for dump file, dump directory, and dump invalid path.

Reply:
done